docs/content/docs/connectors/datastream/kinesis.md

```diff
@@ -566,124 +583,124 @@ Retry and backoff parameters can be configured using the `ConsumerConfigConstant
 
-Users can still switch back to one-thread-per-request mode by setting a key-value pair of `ThreadingModel` and `PER_REQUEST` in `java.util.Properties`, as shown in the code commented out in above example.
+### Kinesis Sinks and Fault Tolerance
 
-### Backpressure
+The sink is designed to participate in Flink's checkpointing to provide at-least-once processing guarantees. It does this by flushing the entire contents of the buffer when a checkpoint reaches the sink. This effectively assures all requests that were triggered before the checkpoint have been successfully acknowledged by Kinesis Data Streams, before proceeding to process more records sent to the sink.
 
-By default, `FlinkKinesisProducer` does not backpressure. Instead, records that
-cannot be sent because of the rate restriction of 1 MB per second per shard are
-buffered in an unbounded queue and dropped when their `RecordTtl` expires.
+In case of a failure or a resharding, data will be written again to Kinesis, leading to duplicates.  Also, the sink does not guarantee that records are written in order to the shards.
 
-To avoid data loss, you can enable backpressuring by restricting the size of the
-internal queue:
+To use fault tolerant Kinesis Sinks, checkpointing of the topology needs to be enabled at the execution environment.
 
-```
-// 200 Bytes per record, 1 shard
-kinesis.setQueueLimit(500);
-```
+### Backpressure
 
-The value for `queueLimit` depends on the expected record size. To choose a good
-value, consider that Kinesis is rate-limited to 1MB per second per shard. If
-less than one second's worth of records is buffered, then the queue may not be
-able to operate at full capacity. With the default `RecordMaxBufferedTime` of
-100ms, a queue size of 100kB per shard should be sufficient. The `queueLimit`
-can then be computed via
+Backpressure in the sink arises as the sink buffer fills up and writes to the sink 
+begins to exhibit blocking behaviour. Kinesis Data Streams has a rate restriction of 
+1 MB per second per shard.
 
+You can ease backpressure by adjusting the size of the internal queue:
 ```
-queue limit = (number of shards * queue size per shard) / record size
+KinesisDataStreamsSink<String> kdsSink =
+    KinesisDataStreamsSink.<String>builder()
+        ...
+        .setMaxBufferedRequests(10_000)
+        ...
 ```
 
-E.g. for 200Bytes per record and 8 shards, a queue limit of 4000 is a good
-starting point. If the queue size limits throughput (below 1MB per second per
-shard), try increasing the queue limit slightly.
+## Kinesis Producer
+
+{{< hint warning >}}
+The Kinesis Producer is deprecated and may be removed with a future release of Flink, please use [Kinesis Sink]({{<ref "docs/connectors/datastream/kinesis#kinesis-data-streams-sink">}}) instead.
+{{< /hint >}}
 
+For older references you can look at the Flink 1.14 documentation.
```

Comment:
I am not sure what this means. Are you referring to the Kinesis Producer Library? 

We should word this section to reflect the Flink class name `FlinkKinesisProducer`. Point out that this legacy sink used the `Kinesis Producer Library (KPL)` whereas the new one does not. We should highlight here the gap; the new sink does not support aggregation